webdriver-spec.html

```diff
@@ -5340,26 +5402,146 @@ <h3 id="character-types">Character types</h3>
+ If the subsequent requested <a>command</a> is not one listed in this chapter,
+ an <a>unexpected alert open</a> <a>error</a> will be returned.
+
+<p><a href=https://html.spec.whatwg.org/#user-prompts>User prompts</a>
+ that are spawned from <code>onbeforeunload</code> event handlers,
+ are <a>dismissed</a> implicitly upon <a href=#>navigation</a>,
+ regardless of the defined <a>user prompt handler</a>.
+
+<p>The following <dfn>table of simple dialogues</dfn> enumerates all
+ supported <a href=https://html.spec.whatwg.org/#simple-dialogs>simple dialogues</a>,
+ along with the <a>commands</a> that are allowed to interact with it
+ as a non-normative reference:
 
-<p class=issue><strong>Warning:</strong>
- This section has not yet been redefined to match the <a>routing requests</a> model,
- and uses old concepts and definitions.
- Please do not rely on it yet.
+<table class=simple>
+ <tr>
+  <th>Definition
+  <th>Dialogue
+  <th>Interactions
+ </tr>
 
-<p>This section describes how modal dialogs
- should be handled using the WebDriver API.
+ <tr>
+  <td><dfn>Alert</dfn>
+  <td>window.<a href=https://html.spec.whatwg.org/#dom-alert><code>alert</code></a>
+  <td>
+   <a>Accept Alert</a><br />
+   <a>Dismiss Alert</a><br />
+   <a>Get Alert Text</a>
+ </tr>
 
-<section>
-<h3>window.alert, window.prompt and window.confirm</h3>
+ <tr>
+  <td><dfn>Confirm</dfn>
+  <td>window.<a href=https://html.spec.whatwg.org/#dom-confirm><code>confirm</code></a>
+  <td>
+   <a>Dismiss Alert</a><br />
+   <a>Accept Alert</a><br />
+   <a>Get Alert Text</a>
+ </tr>
+
+ <tr>
+  <td><dfn>Prompt</dfn>
+  <td>window.<a href=https://html.spec.whatwg.org/#dom-prompt><code>prompt</code></a>
+  <td>
+   <a>Dismiss Alert</a><br />
+   <a>Accept Alert</a><br />
+   <a>Get Alert Text</a><br />
+   <a>Send Alert Text</a>
+ </tr>
+</table>
+
+<p class=issue>It’s not clear what interoperability looks like
+ for the ability to switch top-level browsing contexts
+ whilst a user prompt is present.
+
+<p>The <dfn>current user prompt</dfn> is said to be the active
+ <a href=https://html.spec.whatwg.org/#user-prompts>user prompt</a>,
+ which can be one of the entries on the <a>table of simple dialogues</a>,
+ the window.<a href=https://html.spec.whatwg.org/#dom-print><code>print</code></a> dialogue,
+ or a dialogue spawned by <a href=https://html.spec.whatwg.org/#dom-showmodaldialog><code>showModalDialog()</code></a>
+ should this be supported by the user agent.
+
+<p>To <dfn data-lt="dismiss|dismissed|dismisses">dismiss</dfn>
+ the <a>current user prompt</a>,
+ do so as if the user would click the <b>Cancel</b> or <b>OK</b> button,
+ whichever is present, in that order.
+
+<p>To <dfn data-lt="accept|accepted|accepts">accept</dfn> the <a>current user prompt</a>,
+ do so as if the user would click the <b>OK</b> button.
+
+<p>A <dfn>user prompt handler</dfn> is
+ an <a href=https://html.spec.whatwg.org/#enumerated-attribute>enumerated attribute</a>
```

Comment:
So the idea is to set this user prompt handler when we receive the capabilities, but I couldn’t figure out how to add it in a clean way.  This is the reason I’m using a so-called “enumerated attribute” from HTML and define a string keyword lookup for each handler.